mesonpy/_editable.py

```diff
@@ -18,6 +19,37 @@
 # It is copied during the editable hook installation.
 
 
+_COLORS = {
+    'cyan': '\33[36m',
+    'yellow': '\33[93m',
+    'light_blue': '\33[94m',
+    'bold': '\33[1m',
+    'dim': '\33[2m',
+    'underline': '\33[4m',
+    'reset': '\33[0m',
+}
+_NO_COLORS = {color: '' for color in _COLORS}
+
+
+def _init_colors() -> Mapping[str, str]:
+    """Detect if we should be using colors in the output. We will enable colors
+    if running in a TTY, and no environment variable overrides it. Setting the
+    NO_COLOR (https://no-color.org/) environment variable force-disables colors,
+    and FORCE_COLOR forces color to be used, which is useful for thing like
+    Github actions.
+    """
+    if 'NO_COLOR' in os.environ:
+        if 'FORCE_COLOR' in os.environ:
+            warnings.warn('Both NO_COLOR and FORCE_COLOR environment variables are set, disabling color')
+        return _NO_COLORS
+    elif 'FORCE_COLOR' in os.environ or sys.stdout.isatty():
+        return _COLORS
+    return _NO_COLORS
+
+
+_STYLES = _init_colors()  # holds the color values, should be _COLORS or _NO_COLORS
+
+
```

Comment:
Do we really need to duplicate all this here and in `__init__`?

Reply:
This module is meant to be standalone, it gets copied to the editable hook.